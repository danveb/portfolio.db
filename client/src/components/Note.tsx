import { useState, useRef } from "react";
import { MessageSent } from "./index";
import emailjs from "@emailjs/browser";
import "../styles/Note.css";

const serviceId = process.env.REACT_APP_SERVICE_ID;
const templateId = process.env.REACT_APP_TEMPLATE_ID;
const publicKey = process.env.REACT_APP_PUBLIC_KEY;

export default function Note() {
  // useState
  const [messageSent, setMessageSent] = useState<boolean>(false);

  const form = useRef<HTMLFormElement | null>(null);
  const sendEmail = (e: React.FormEvent) => {
    e.preventDefault();

    // error check for .env on EmailJS
    if (!serviceId || !templateId || !publicKey) {
      throw new Error("Environment variables not set correctly.");
    }
    emailjs.sendForm(serviceId, templateId, form.current as HTMLFormElement, publicKey)
      .then((result) => {
        console.log(result.text);
        setMessageSent(() => !messageSent);
      }, (error) => {
        console.log(error.text);
      });
  };

  return (
    <div className="note">
      <div className="note__wrapper">
        {messageSent ? (
          <MessageSent />
        ) : (
          <form ref={form} onSubmit={sendEmail}>
            <label htmlFor="user_name">Name</label>
            <input
              type="text"
              id="user_name"
              name="user_name"
              placeholder="Your name"
              required
            />
            <label htmlFor="user_email">Email</label>
            <input
              type="email"
              id="user_email"
              name="user_email"
              placeholder="Your email"
              required
            />
            <label htmlFor="message">Message</label>
            <textarea
              name="message"
              id="message"
              placeholder="Any message you'd like to send"
              required
            />
            <input className="submitBtn" type="submit" value="shoot email 👨‍💻" />
          </form>
        )}
      </div>
    </div>
  );
}