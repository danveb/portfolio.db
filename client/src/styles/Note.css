.note {
  width: 100%; 
  height: 100%; 
}

.note__wrapper {
  padding: 15px 20px; 
  background-color: var(--raisin-black);
  border-radius: 10px; 
}

form {
  display: flex; 
  flex-direction: column; 
  justify-content: center; 
  gap: 15px; 
  color: var(--alice-blue); 
  font-family: "Mulish", sans-serif; 
  font-size: 14px; 
}

label, 
input {
  font-family: inherit; 
  font-size: inherit; 
}

input, 
textarea {
  padding: 10px; 
  outline: none; 
  font-family: inherit; 
  resize: none; 
}

.submitBtn {
  background-color: var(--rich-black); 
  border: none; 
  color: #fff; 
  cursor: pointer; 
}