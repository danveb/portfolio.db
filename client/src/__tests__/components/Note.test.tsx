import { render, screen } from "@testing-library/react"; 
import Note from "../../components/Note"; 
import { BrowserRouter } from "react-router-dom";

describe("Note component", () => {
  // smoke test
  test("renders component wtesthout crashing", () => {
    render(
      <BrowserRouter>
        <Note />
      </BrowserRouter>
    ); 
  });

  // snapshot test 
  test("matches snapshot", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Note />
      </BrowserRouter>
    ); 
    expect(asFragment()).toMatchSnapshot(); 
  }); 

  // getByLabelText
  test("name, email and message labels are properly displayed on contact form", () => {
    render(
      <BrowserRouter>
        <Note />
      </BrowserRouter>
    ); 
    const nameLabel = screen.getByLabelText("Name"); 
    const emailLabel = screen.getByLabelText("Email"); 
    const messageLabel = screen.getByLabelText("Message"); 
    expect(nameLabel).toBeInTheDocument(); 
    expect(emailLabel).toBeInTheDocument(); 
    expect(messageLabel).toBeInTheDocument(); 
  }); 
})