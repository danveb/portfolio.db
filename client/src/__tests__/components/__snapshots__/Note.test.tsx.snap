// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="note"
  >
    <div
      class="note__wrapper"
    >
      <form>
        <label
          for="user_name"
        >
          Name
        </label>
        <input
          id="user_name"
          name="user_name"
          placeholder="Your name"
          required=""
          type="text"
        />
        <label
          for="user_email"
        >
          Email
        </label>
        <input
          id="user_email"
          name="user_email"
          placeholder="Your email"
          required=""
          type="email"
        />
        <label
          for="message"
        >
          Message
        </label>
        <textarea
          id="message"
          name="message"
          placeholder="Any message you'd like to send"
          required=""
        />
        <input
          class="submitBtn"
          type="submit"
          value="shoot email 👨‍💻"
        />
      </form>
    </div>
  </div>
</DocumentFragment>
`;
